src/pk.c

```diff
@@ -3556,13 +3556,16 @@ int wolfSSL_RSA_verify_PKCS1_PSS(WOLFSSL_RSA *rsa, const unsigned char *mHash,
 
     if (ret == 1) {
         /* Calculate the salt length to use for special cases. */
-        /* TODO: use special case wolfCrypt values. */
         switch (saltLen) {
         /* Negative saltLen values are treated differently */
         case RSA_PSS_SALTLEN_DIGEST:
             saltLen = hashLen;
             break;
-        case RSA_PSS_SALTLEN_MAX_SIGN:
+        case RSA_PSS_SALTLEN_AUTO:
+        #ifdef WOLFSSL_PSS_SALT_LEN_DISCOVER
+            saltLen = RSA_PSS_SALT_LEN_DISCOVER;
+            break;
+        #endif
```

Comment:
Perhaps the `#else` case should return an error?

Reply:
No case now when WOLFSSL_PSS_SALT_LEN_DISCOVER not defined.
Will go to default which errors out on negative.

Reply:
I'd really like a test case for this... Could we add one?

Reply:
Actually that doesn't work.
-2 is  RSA_PSS_SALTLEN_MAX_SIGN for signing and using -2 should have the same semantics in verify.
Discovery will handle anything.